python/tvm/relay/qnn/op/legalizations.py

```diff
@@ -91,12 +91,30 @@ def qnn_conv2d_transpose_legalize(attrs, inputs, types):
     # Collect the input exprs.
     data, kernel, input_zero_point, kernel_zero_point, _, _ = inputs
 
-    shift_data = relay.subtract(
```

Comment:
subtract should already have numpy style broadcasting so I'm not sure you need to replace the things here,

Reply:
Without this change, the model I'm looking at fails with `Incompatible broadcast type TensorType([32, 32, 3, 3], int16) and TensorType([32], int16)`. Here is the op for more info:
```
  %2181 = qnn.conv2d_transpose(%2180, meta[relay.Constant][1003] /* ty=Tensor[(32, 32, 3, 3), int8] */, -26 /* ty=int32 */, meta[relay.Constant][1004] /* ty=Tensor[(32), int32] */, 0.196949f /* ty=float32 */, meta[relay.Constant][1005] /* ty=Tensor[(32), float32] */, channels=32, kernel_size=[3, 3], strides=[2, 2], padding=[0, 0, 1, 1], kernel_layout="IOHW", out_dtype="int32") /* ty=Tensor[(1, 32, 100, 100), int32] */;
```

Reply:
Ah I see yeah that should be correct, simple arithmetic should follow generic broadcasting rules: https://numpy.org/doc/stable/user/basics.broadcasting.html#general-broadcasting-rules.

Reply:
I think the preferable way is to use `relay.broadcast_to_like` or `relay.broadcast_to`. It looks like bias_add `include/tvm/topi/nn/bias_add.h` makes some additional assumptions about the input shape.

Reply:
I tried it but got "incompatible broadcast type" [32] --> [32, 32, 3, 3] down the line. I was actually looking at how this was handled in `helper_no_fast_int8_hw_legalization` and it seems `biad_add` is the preferred way?

Reply:
Hmm ok. that's weird